src/repl/scala/tools/nsc/interpreter/IMain.scala

```diff
@@ -105,6 +105,29 @@ class IMain(initialSettings: Settings, protected val out: JPrintWriter) extends
     try body
     finally if (!saved) settings.nowarn.value = false
   }
+  def withSuppressedSettings[A](body: => A): A = {
```

Comment:
Can we avoid the duplication with https://github.com/scala/scala/blob/v2.12.2/src/repl/scala/tools/nsc/interpreter/ILoop.scala#L958 ?